vignettes/generation.Rmd

```diff
@@ -0,0 +1,170 @@
+---
+title: "Generating qseaSets"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Generating qseaSets}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(mesa)
+library(stringr)
+library(tibble)
+```
+
+# Introduction
+In this vignette we cover the generation of a qseaSet using the functions provided in `mesa`, rather than those in `qsea`. Please see the [`qsea` tutorial](https://bioconductor.org/packages/devel/bioc/vignettes/qsea/inst/doc/qsea_tutorial.html) for more detail, but we will cover much of that here.
+
+## Sample Metadata 
+The first thing to prepare is the `sampleTable`; a dataframe containing metadata for the samples used in the package. This has three required columns:
+
+* `sample_name` : a name for each sample, which must be unique. This is the primary identifier used to describe each sample, so it is good to be descriptive.
+* `group` : a column to specify a group for each sample, for use inside certain functions for averaging over samples of the same type. We suggest only using this for replicates of the same (or very similar) samples, and adding extra columns for other comparisons that may be of interest.
+* `file_name` : A path to a bam file, containing the aligned, deduplicated, reads mapped to the relevant reference genome, from material which has undergone the methylation enrichment/capture process (e.g. MEDIP-seq or MBD-seq). Ideally `samtools fixmate` has been ran to supply mate information onto the read tags, but filtering for quality is not required.
+
+There are also two more optional fields that may be used during generation:
+
+* `input_file` : This is a path to a bam file containing aligned, deduplicated reads for material from the same sample, without performing enrichment. This is whole genome sequencing, but may be sequenced at a much lower depth (e.g. ultra-shallow WGS, ~1-2 million reads). If present, this will be used on a sample-by-sample basis to calculate copy number variation, at a chromosome scale, in order to correct for this in the expected read depths. This is most relevant when studying cancer, as particular cancers often show recurrant patterns of copy number changes. 
```

Comment:
Might be worth quoting references that use different depths, or outline briefly what can't and can be done with swGS-like depths compared to WGS (or if there's relevant literature, point to the reference, or point to `qsea` if it's discussed there). 
IIRC the published papers in which `mesa` is used have 30-50MM reads per sample, from the supplementary materials I've read.

Reply:
> IIRC the published papers in which `mesa` is used have 30-50MM reads per sample, from the supplementary materials I've read.

Those are the values for the enriched fraction, the non-enriched fraction generally has ~10x fewer reads. Compare column J and column O of the [supplementary information](https://static-content.springer.com/esm/art%3A10.1038%2Fs43018-022-00415-9/MediaObjects/43018_2022_415_MOESM2_ESM.xlsx) for the Nature Cancer paper.

Reply:
Thanks, I must have missed that.